zebrad/tests/acceptance.rs

```diff
@@ -887,9 +903,28 @@ fn sync_until(
     if enable_mempool_at_height.is_some() {
         child.expect_stdout_line_matches("enabling mempool for debugging")?;
         child.expect_stdout_line_matches("activating mempool")?;
+
+        // make sure zebra is running with the mempool
+        child.expect_stdout_line_matches("verified checkpoint range")?;
     }
 
     child.expect_stdout_line_matches(stop_regex)?;
+
+    // make sure there is never a mempool if we don't explicity enable it
+    if enable_mempool_at_height.is_none() {
+        // if there is no matching line, the `expect_stdout_line_matches` error kills the `zebrad` child.
+        // the error is delayed until the test timeout, or until the child reaches the stop height and exits.
+        let mempool_is_activated = child
+            .expect_stdout_line_matches("activating mempool")
+            .is_ok();
+
+        // if there is a matching line, we panic and kill the test process.
+        // but we also need to kill the `zebrad` child before the test panics.
+        if mempool_is_activated {
+            child.kill()?;
+            panic!("unexpected mempool activation: mempool should not activate while syncing lots of blocks")
+        }
+    }
     child.kill()?;
```

Comment:
I was wrong about calling `kill` multiple times.

Here's the fix to ignore the "can't kill an exited process" error.

```suggestion

    // make sure the child process is dead
    // if it has already exited, ignore that error
    let _ = child.kill();
```